addons/web_hierarchy/static/src/hierarchy_model.js

```diff
@@ -670,29 +676,65 @@ export class HierarchyModel extends Model {
         config.context = "context" in params ? params.context : config.context;
         if ("domain" in params) {
             config.domain = params.domain;
-            if (!params.domain.length) {
-                if (config.context.hierarchy_res_id) {
```

Comment:
Coucou @awa-odoo,

Indeed, I agree, it is a bit magic but if we do what you suggest we have to do it with `id` field (that is, `search_default_id`) and so it could be weird to have `id: 1` in the search view since the id is more technical, IMO.
But I am not against to do that if everyone is okay to have that in the search view. 🙂

Reply:
Coucou 👋 

Indeed, I think you can rename the search label but you would still have "Employee ID: 42", which is not great visually.

As an additional note, I was also a bit confused by the fact that we disregard that context when there is any other filter.

Say you want to view the Org Chart of the Employee "Jeffrey Kelly":
![image](https://github.com/odoo/odoo/assets/41282536/5c25189b-d354-4707-acd9-fac20e041824)

Wouldn't you want to be able to further refine the Search, for example to be able to see all the Consultant under his level?
Currently if you do that, you see all the Consultants, which seems a bit strange to me:
![image](https://github.com/odoo/odoo/assets/41282536/c39328ec-0788-42f5-a746-c48c85267a81)

Using the default search as suggested would have enabled that, and also allowed the user to remove "Jeffrey Kelly" from the filter if you want to see everyone instead.

This is more of a functional choice though so maybe something to discuss with GMF, we don't have this kind of features in Knowledge (currently) so it does not really affect us, just wanted to ask.
(It can also of course be done/discussed later, not needed for the freeze).

Thanks for all the answers & precisions.

Cheers!